pysmt/solvers/portfolio.py

```diff
@@ -70,6 +71,7 @@ def __init__(self, solvers_set, environment, logic, **options):
 
         One process will be used for each of the solvers.
         """
+        logic = convert_logic_from_string(logic)
```

Comment:
Would it make sense to accept both a `str` or a `Logic` as `logic`?

Something like:
```python
if isinstance(logic, str):
   logic = convert_logic_from_string(logic)
```

Reply:
That is what the helper does internally. Maybe the name should make it more explicit:
```python
def convert_logic_from_string(name):
    """Helper function to parse function arguments.

    This takes a logic or a string or None, and returns a logic or None.
    """
    if name is not None and isinstance(name, str):
        name = get_logic_by_name(name)
    return name
```